src/gourmand/backends/db.py

```diff
@@ -2122,22 +2120,20 @@ def get_database (*args, **kwargs):
     return RecData.instance_for(*args, **kwargs)
 
 
-def backup_database(filename: Path) -> Path:
-    backup_name = filename.with_name(filename.name + '.backup-' + time.strftime('%d-%m-%y'))
+def backup_database(filename: Union[None, Path, str]) -> Union[None, Path]:
+    if filename is None:
+        return None
+
+    if type(filename) == str:
+        filename = Path(filename)
+
+    backup_name = filename.with_name(
+        filename.name + '.backup-' + time.strftime('%d-%m-%y')
+    )
 
     while backup_name.is_file():
         backup_name = backup_name.with_name(backup_name.name + 'I')
 
-    show_message(
```

Comment:
Todo:

- [ ] As [noted](https://github.com/GourmandRecipeManager/gourmand/issues/137#issuecomment-1462692210), this informational message needs to be restored elsewhere.

Reply:
A more-relevant hyperlink about the informational message: https://github.com/MrShark/gourmand/commit/b6b8d18854585bdc50cb3338f49bfeab67cf49af

Reply:
What do you think about the approach taken in feb0d5a0807a6393ce893d535e79fc90d9ea29fc @cydanil @MrShark?

It uses a SQLite 'in-memory' URI for the database.  That means that `self.filename` for the database is empty, and the backup is skipped.

It feels a little fragile - it's relying on a series of interconnected things fitting together.  But it works (tests pass, and the UI doesn't display the 'backup' message -- but should still do during interactive usage when a database upgrade occurs).

Reply:
(I'd probably like to attempt to find a better solution, but until then.. review & feedback welcome)